dev/NavigationView/NavigationView.cpp

```diff
@@ -1526,31 +1527,58 @@ void NavigationView::UpdatePaneLayout()
+                                    }
                                 }
-                                return menuItemsActualHeight;
+                                return menuItemsActualHeight - separatorLineActualHeight;
                             }
                             else if (footersActualHeight <= totalAvailableHeightHalf)
                             {
                                 // Menu items exceed over the half, so let's limit them.
                                 footerItemsScrollViewer.MaxHeight(footersActualHeight);
+                                double separatorLineActualHeight = 0.0;
                                 if (const auto& separator = m_visualItemsSeparator.get())
                                 {
-                                    separator.Visibility(winrt::Visibility::Visible);
+                                    if (footersActualHeight > 0)
+                                    {
+                                        separator.Visibility(winrt::Visibility::Visible);
+                                        separatorLineActualHeight = separatorLineHeight;
+                                    }
+                                    else
+                                    {
+                                        separator.Visibility(winrt::Visibility::Collapsed);
+                                    }
                                 }
-                                return totalAvailableHeight - footersActualHeight;
+                                return totalAvailableHeight - footersActualHeight - separatorLineActualHeight;
                             }
                             else
                             {
                                 // Both are more than half the height, so split evenly.
                                 footerItemsScrollViewer.MaxHeight(totalAvailableHeightHalf);
+                                double separatorLineActualHeight = 0.0;
                                 if (const auto& separator = m_visualItemsSeparator.get())
                                 {
-                                    separator.Visibility(winrt::Visibility::Visible);
+                                    if (footersActualHeight > 0)
+                                    {
+                                        separator.Visibility(winrt::Visibility::Visible);
+                                        separatorLineActualHeight = separatorLineHeight;
+                                    }
+                                    else
+                                    {
+                                        separator.Visibility(winrt::Visibility::Collapsed);
+                                    }
                                 }
-                                return totalAvailableHeightHalf;
+                                return totalAvailableHeightHalf - separatorLineActualHeight;
```

Comment:
@StephenLPeters @ranjeshj @karenbtlai this part with setting visibility of separator as well as actual height repeats 3 times - should I extract it somewhere down below?